syscalls.c

```diff
@@ -261,12 +261,45 @@ int syscalls_endthread(void *ustack)
 }
 
 
-int syscalls_usleep(void *ustack)
+int syscalls_nsleep(void *ustack)
 {
-	unsigned int us;
+	time_t *sec;
+	long int *nsec;
+	time_t start, sleep, stop, diff;
+	int ret;
 
-	GETFROMSTACK(ustack, unsigned int, us, 0);
-	return proc_threadSleep((unsigned long long)us);
+	GETFROMSTACK(ustack, time_t *, sec, 0);
+	GETFROMSTACK(ustack, long int *, nsec, 1);
+
+	/* TODO - user pointer check */
+	if (sec == NULL || nsec == NULL) {
+		return -EINVAL;
+	}
+
+	/* FIXME - time_t should be signed and we should check for *sec < 0 */
+	if (*nsec < 0 || *nsec >= 1000 * 1000 * 1000) {
+		return -EINVAL;
+	}
+
+	proc_gettime(&start, NULL);
+
+	sleep = *sec * 1000 * 1000 + (*nsec + 999) / 1000;
```

Comment:
this doesn't fix the problem with overflowing microseconds when `sleep(4295)` is being used, can this be taken care of (probably better in kernel to avoid ^C^V in libphoenix `nanosleep` and `sleep` implementations)

see: https://github.com/phoenix-rtos/libphoenix/issues/93

Reply:
edit: the `proc_threadSleep` takes `unsigned long long us` as a param, so maybe this is fixed. Please use the expected type directly, don't assume `time_t` would be wide enough.

Reply:
Hmmm, I think this indeed solves it already - both `*sec` and `sleep` are of `time_t` type (`unsigned long long`).

Reply:
Done

Reply:
yeah, confused by the types... but `time_t` should probably be not arch-dependant for it to be always true. Still `proc_threadSleep` should probably take `time_t` as a param (or do we have the convention to use `time_t` for absolute times and `unsigned long long` for periods? If so, some types should be changed here ;))

Reply:
I agree that there is no point for time_t being platform dependent, but this is outside of this PR scope.

Also, it should be shared between kernel and libc and be signed, all these things should be fixed in the future.